packages/core/src/lib/register.ts

```diff
@@ -159,13 +166,13 @@ export const registerWithSocial = async (
 ) => {
   const userInfo = await getUserInfoByConnectorCode(connectorId, code);
 
-  assertThat(
-    !(await hasUserWithIdentity(connectorId, userInfo.id)),
-    new RequestError({
+  if (await hasUserWithIdentity(connectorId, userInfo.id)) {
+    await assignRegistrationResult(ctx, provider, undefined, { connectorId, userInfo });
```

Comment:
do we still need to`assignRegistrationResult` even if userInfo exists?

Reply:
user in UI can reuse this userinfo that "assigned" to session, because `code` is one-time.